### PR TITLE
Add unit test support for EFR32

### PR DESCRIPTION
**Pull request by rgoliver**
#### Problem
There is no way to run the unit tests on EFR32.

#### Change overview
- Add an on device test driver for EFR32, this uses RPCs which are actuated using the included python test runner.
- Fix a subset of the unit tests so they work with EFR32 and include them in the build. 
- Included a simple python test runner script which can be used to flash and run the tests.

#### Testing
Built and ran using the included test runner on an EFR32MG12 board, passes all of the included 265 tests.

### CODE REVIEW

**Review by jmartinez-silabs — APPROVED** [review]

**Comment by andy31415** [comment]
@mspang for build system changes

**Review by mspang — APPROVED** [review]

File: src/lib/support/UnitTestUtils.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+#pragma once
+
+#include <stdint.h>
+
+namespace chip {
+namespace test_utils {
+
+void sleep_us(uint64_t microsecs);
```
[the remark this answers is not in the document]
Comment (rgoliver) [reply src/lib/support/UnitTestUtils.h]: Done

File: build/chip/tests.gni [hunk 2]
```diff
@@ -28,6 +28,10 @@ declare_args() {
   # Build executables for running individual tests.
   chip_link_tests =
       chip_build_tests && (current_os == "linux" || current_os == "mac")
+
+  # Use source_set instead of static_lib for tests.
+  chip_use_source_set_for_tests =
```
[the remark this answers is not in the document]
Comment (rgoliver) [reply build/chip/tests.gni]: Done

**Review by bzbarsky-apple — APPROVED** [review]

File: build/chip/tests.gni [hunk 3]
```diff
@@ -28,6 +28,10 @@ declare_args() {
   # Build executables for running individual tests.
   chip_link_tests =
       chip_build_tests && (current_os == "linux" || current_os == "mac")
+
+  # Use source_set instead of static_lib for tests.
+  chip_build_test_static_libraries =
+      !(chip_build_tests || chip_device_platform == "efr32")
```
Comment (mspang) [inline build/chip/tests.gni:34]: Shouldn't this just be

`chip_device_platform != "efr32"`

Comment (rgoliver) [reply build/chip/tests.gni]: Done

**Pull request merged**